auth/oidc.go

```diff
@@ -82,12 +102,36 @@ type OIDCOptions struct {
 // with an OpenID Connect provider.
 type OIDCClient struct {
 
+	// configLock is a reader/writer mutual exclusion lock on config.
+	configLock sync.RWMutex
+
 	// Config describes a typical OAuth2 flow, with both the client
 	// application information and the server's endpoint URLs.
-	Config oauth2.Config
+	config *oauth2.Config
+
+	// verifierLock is a reader/writer mutual exclusion lock on verifier.
+	verifierLock sync.RWMutex
+
+	// verifier provides verification for ID Tokens.
+	verifier *oidc.IDTokenVerifier
+}
+
+func (client *OIDCClient) SetVerifier(verifier *oidc.IDTokenVerifier) {
+	client.verifierLock.Lock()
+	client.verifier = verifier
+	client.verifierLock.Unlock()
+}
+
+func (client *OIDCClient) Config() *oauth2.Config {
+	client.configLock.Lock()
+	defer client.configLock.Unlock()
+	return client.config
+}
 
-	// Verifier provides verification for ID Tokens.
-	Verifier *oidc.IDTokenVerifier
+func (client *OIDCClient) SetEndpoint(endpoint oauth2.Endpoint) {
+	client.configLock.Lock()
+	client.config.Endpoint = endpoint
+	client.configLock.Unlock()
```

Comment:
This (and Config() above) don't look like they are actually providing any race protection for client.config.Endpoint.  A caller that invokes Config() is going to get a reference to an oauth2Config, and can then attempt to read config.Endpoint without holding the mutex.  SetEndpoint can concurrently attempt to update the endpoint value on the same oauth2Config.